src/Eventso.Subscription.Kafka/DeadLetter/PoisonEventQueue.cs

```diff
@@ -0,0 +1,166 @@
+using System.Collections.Concurrent;
+using System.Runtime.CompilerServices;
+using Confluent.Kafka;
+using Microsoft.Extensions.Logging;
+
+namespace Eventso.Subscription.Kafka.DeadLetter;
+
+public sealed class PoisonEventQueue(
+    IPoisonEventStore poisonEventStore,
+    IPoisonEventRetryScheduler retryScheduler,
+    string groupId,
+    int maxNumberOfPoisonedEventsInTopic,
+    ILogger<PoisonEventQueue> logger)
+    : IPoisonEventQueue
+{
+    private readonly ConcurrentDictionary<TopicPartition, TopicPartitionKeysTask> _partitionPoisonKeys =
+        new ConcurrentDictionary<TopicPartition, TopicPartitionKeysTask>(concurrencyLevel: 1, capacity: 1);
+
+    public bool IsEnabled => true;
+
+    public void Assign(TopicPartition topicPartition)
+    {
+        // no concurrency, same thread as consume
+        var tokenSource = new CancellationTokenSource();
+        _partitionPoisonKeys[topicPartition] = new TopicPartitionKeysTask(
+            GetPoisonKeys(tokenSource.Token),
```

Comment:
let's Task.Run or yield to avoid any additional work on consume loop

Reply:
++